### PR TITLE
Create health check for PrometheusService

### LINKED ISSUE #1207: Create health check for PrometheusService
As part of #1183, all services should maintain a Status method that will return an error if there is any unhealthy status to report. This task requires some creative thinking to evaluate a definition of a healthy service. It is possible that a service can always be healthy and in that case we would simply remove the TODO and update the godoc to explain why the service is always healthy.

Service definition: shared/prometheus/service.go

What makes this service healthy or unhealthy?

Ideas:
- Report unhealthy if the prometheus service is not serving HTTP traffic for `/metrics` or `/healthz`

### PR DESCRIPTION
**Pull request by cyberbono3**
Resolves #1207

### CODE REVIEW

File: shared/prometheus/service.go [hunk 1]
```diff
@@ -93,5 +95,9 @@ func (s *Service) Stop() error {
 // Status always returns nil.
 // TODO(1207): Add service health checks.
 func (s *Service) Status() error {
+	if s.failStatus != nil {
```
[the remark this answers is not in the document]
Comment (prestonvanloon) [reply shared/prometheus/service.go]: +1, A test would be nice, and fairly straightforward

[the remark this answers is not in the document]
Comment (cyberbono3) [reply shared/prometheus/service.go]: Sure .Added TestStatus()

**Comment by rauljordan** [comment]
Hey @cyberbono3 you have some failures due to some undefined variable - please fix so we can proceed with merge, thanks!

File: shared/prometheus/service.go [hunk 2]
```diff
@@ -92,5 +94,9 @@ func (s *Service) Stop() error {
...
+		log.Errorf("Prometheus service is unhealthy : %s", s.server.Addr)
```
Comment (prestonvanloon) [inline shared/prometheus/service.go]: Please remove this. The `Status()` method is called many times and this will bloat the logs

Comment (cyberbono3) [reply shared/prometheus/service.go]: You mean remove line 98 or both lines (97 and 98)?

**Comment by cyberbono3** [comment]
> Hey @cyberbono3 you have some failures due to some undefined variable - please fix so we can proceed with merge, thanks!

What undefined variable do you mean? could you give me more info please?

File: shared/prometheus/service_test.go [hunk 3]
```diff
@@ -84,3 +84,11 @@ func TestHealthz(t *testing.T) {
 	}
 
 }
+
+func TestStatus(t *testing.T) {
+	fStatus:= s.failStatus
```
Comment (nisdas) [inline shared/prometheus/service_test.go]: @cyberbono3 There is no `s` here , so you are getting a compile error

Comment (prestonvanloon) [reply shared/prometheus/service_test.go]: How about you do this:
```go
failure := "something really bad happened"
s := &Status{failStatus: failure}

...
```

**Pull request closed without merging**